plugins/blockvault_client_plugin/blockvault_client_plugin.cpp

```diff
@@ -20,27 +21,33 @@ class blockvault_client_plugin_impl : public vault_impl {
 
 blockvault_client_plugin::blockvault_client_plugin() {}
 
-blockvault_client_plugin::~blockvault_client_plugin() {
-}
+blockvault_client_plugin::~blockvault_client_plugin() {}
 
 void blockvault_client_plugin::set_program_options(options_description&, options_description& cfg) {
 #ifdef HAS_PQXX
    cfg.add_options()("block-vault-backend", bpo::value<std::string>(),
-                     "the url for block vault backend. Currently, only PostgresQL is supported, the format is "
+                     "the uri for block vault backend. Currently, only PostgresQL is supported, the format is "
                      "'postgresql://username:password@localhost/company'");
 #endif
 }
 
 void blockvault_client_plugin::plugin_initialize(const variables_map& options) {
+   ilog("initializing blockvault_client plugin");
 #ifdef HAS_PQXX
-   if (options.count("block-vault-backend")) {
-      std::string url = options["block-vault-backend"].as<std::string>();
-      if (boost::starts_with(url, "postgresql://")) {
-         my.reset(new blockvault_client_plugin_impl(std::make_unique<blockvault::postgres_backend>(url)));
-      } else if (url.size()) {
-         elog("unknown block-vault-backend option, skipping it");
+   try {
+      if (options.count("block-vault-backend")) {
+         std::string uri = options["block-vault-backend"].as<std::string>();
+         if (boost::starts_with(uri, "postgresql://") || boost::starts_with(uri, "postgres://")) {
+            my.reset(new blockvault_client_plugin_impl(std::make_unique<blockvault::postgres_backend>(uri)));
+            ilog("blockvault_client plugin started");
+         } else if (uri.size()) {
+            elog("unknown block-vault-backend option, skipping it");
+         }
       }
    }
+   catch (std::exception& ex) {
+      elog("blockvault_client_plugin initialization error: $msg, skipping it.", ("msg", ex.what()));
```

Comment:
Here and above. If not configured correctly I think you should throw an exception and not allow nodeos to start. This silent failure seems like an issue. If you asked for blockvault you should get it or nodeos should not start.